src/main/java/editpendinglist/EditPendingListDsGateway.java

```diff
@@ -0,0 +1,33 @@
+package editpendinglist;
+
+import Entities.Group;
+import Entities.User;
+
+import java.util.HashMap;
+
+public interface EditPendingListDsGateway {
+    User getUser(String username);
+    Group getGroup(String groupName);
+
+    /**
+     * @return a hashmap of all saved users, mapping username to user object
+     */
+    HashMap<String, User> getUserMap();
+    // I'm not sure what the parameters for the 2 methods below should be, they might be response models
+
+    /**
+     * Updates the repository with the accepted/rejected user's updated group information
+     * @param username the user's username
+     */
+    void updateUser(String username);
+
+    /**
+     * Updates the repository with the group's updated member information
+     * @param groupName the group's name
+     */
+    void updateGroup(String groupName);
+    boolean userInGroup(String username, String groupName);
+    boolean groupInUser(String groupName, String username);
```

Comment:
Hello, I was wondering why you were using the userInGroup and groupInUser methods. I couldn't see them used in the interactor class. I was wondering if they were necessary for the code to be concise. If they are, it would be great if you explained which would allow me to implement it for my use case as well?

Reply:
Yeah these methods are just for my testing which I haven't implemented yet. For example, I need to check if my interactor actually added a user to a group in my test file so I'll use these then.